src/components/views/auth/InteractiveAuthEntryComponents.js

```diff
@@ -564,6 +600,88 @@ export const MsisdnAuthEntry = createReactClass({
     },
 });
 
+export class SSOAuthEntry extends React.Component {
+    static propTypes = {
+        matrixClient: PropTypes.object.isRequired,
+        authSessionId: PropTypes.string.isRequired,
+        loginType: PropTypes.string.isRequired,
+        submitAuthDict: PropTypes.func.isRequired,
+        errorText: PropTypes.string,
+        onPhaseChange: PropTypes.func.isRequired,
+        continueText: PropTypes.string,
+        continueKind: PropTypes.string,
+        onCancel: PropTypes.func,
+    };
+
+    static LOGIN_TYPE = "m.login.sso";
+    static UNSTABLE_LOGIN_TYPE = "org.matrix.login.sso";
+
+    static PHASE_PREAUTH = 1; // button to start SSO
+    static PHASE_POSTAUTH = 2; // button to confirm SSO completed
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            // We actually send the user through fallback auth so we don't have to
+            // deal with a redirect back to us, losing application context.
+            ssoUrl: props.matrixClient.getFallbackAuthUrl(
+                this.props.loginType,
+                this.props.authSessionId,
+            ),
```

Comment:
Does this need to be state? Looks like this is the only place it's calculated.

Reply:
probably not